src/code/z_demo.c

```diff
@@ -1,99 +1,1634 @@
+                Quake_SetSpeed(sCutsceneQuakeIndex, 30000);
+                Quake_SetQuakeValues(sCutsceneQuakeIndex, 20, 10, 0, 0);
+                Quake_SetCountdown(sCutsceneQuakeIndex, 800);
+            }
+            break;
+        case 0x1B:
+            if (isStartFrame) {
+                globalCtx->nextEntranceIndex = 0x1C00;
+                gSaveContext.nextCutsceneIndex = 0xFFF8;
+                globalCtx->sceneLoadFlag = 0x14;
+                globalCtx->unk_1887F = 3;
+            }
+            break;
+        case 0x1C:
+            if (isStartFrame) {
+                globalCtx->envCtx.unk_17 = 0xD;
+            }
+            break;
+        case 0x1D:
+            gSaveContext.playerForm = sCutsceneStoredPlayerForm;
+            break;
+        case 0x1E:
+            D_801F4DE0 = true;
+            break;
+        case 0x1F:
+            D_801F4DE0 = false;
+            break;
+        case 0x21:
+            if (isStartFrame) {
+                func_80146EE8(globalCtx);
+            }
+            break;
+        case 0x22:
+            if (isStartFrame) {
+                func_80144A94(&globalCtx->sramCtx);
+            }
+            break;
+        case 0x23:
+            if (csCtx->frames != D_801BB15C) {
+                D_801BB15C = csCtx->frames;
+
+                if (REG(15) != 0) {
+                    time = gSaveContext.time;
+                    gSaveContext.time = (u16)REG(15) + time;
+                    time = gSaveContext.time;
+                    gSaveContext.time = (u16)gSaveContext.unk_14 + time;
+                }
+            }
+            break;
+        case 0x24:
+            func_8019F128(NA_SE_EV_EARTHQUAKE_LAST - SFX_FLAG);
+            if (isStartFrame) {
+                sCutsceneQuakeIndex = Quake_Add(GET_ACTIVE_CAM(globalCtx), 6);
+                Quake_SetSpeed(sCutsceneQuakeIndex, 22000);
+                Quake_SetQuakeValues(sCutsceneQuakeIndex, 2, 1, 0, 0);
+                Quake_SetCountdown(sCutsceneQuakeIndex, 800);
+            }
+            break;
+
+        case 0x26:
+            gSaveContext.day = 9;
+
+            {
+                GameState* gameState = &globalCtx->state;
+                gameState->running = false;
+            }
+            SET_NEXT_GAMESTATE(&globalCtx->state, Daytelop_Init, DaytelopContext);
+
+            func_80146F5C(globalCtx);
+            break;
+
+        case 0x27:
+            gSaveContext.playerForm = PLAYER_FORM_ZORA;
+            break;
+
+        case 0x28:
+            csCtx->frames = cmd->startFrame - 1;
+            break;
+    }
+}
+
+// Command 0x97: Set Environment Lighting
+void Cutscene_Command_SetLighting(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdEnvLighting* cmd) {
+    if (csCtx->frames == cmd->startFrame) {
+        if (cmd->setting != 0x20) {
+            globalCtx->envCtx.unk_C3 = cmd->setting - 1;
+            globalCtx->envCtx.unk_DC = 1.0f;
+        } else {
+            globalCtx->envCtx.unk_C3 = 0xFF;
```

Comment:
```suggestion
            globalCtx->envCtx.lightSettingOverride = cmd->setting - 1;
            globalCtx->envCtx.lightBlend = 1.0f;
        } else {
            globalCtx->envCtx.lightSettingOverride = 0xFF;
```
This one is entirely optional. But if you wanted to get ahead of the curve from Fig's Environment docs PR...